ocs_ci/deployment/baremetal.py

```diff
@@ -807,6 +817,439 @@ def set_pxe_boot_and_reboot(self, machine):
             run_cmd(cmd=cmd, secrets=secrets)
 
 
+class BAREMETALAI(BAREMETALBASE):
+    """
+    A class to handle Bare metal Assisted Installer specific deployment
+    """
+
+    def __init__(self):
+        logger.info("BAREMETAL AI")
+        super(BAREMETALAI, self).__init__()
+
+    class OCPDeployment(BMBaseOCPDeployment):
+        def __init__(self):
+            super(BAREMETALAI.OCPDeployment, self).__init__()
+
+        def deploy_prereq(self):
+            """
+            Pre-Requisites for Bare Metal AI Deployment
+            """
+            super().deploy_prereq()
+
+            # create initial metadata.json file in cluster dir, to ensure, that
+            # destroy job will be properly triggered even when the deployment fails
+            # and metadata.json file will not be created
+            with open(
+                os.path.join(self.cluster_path, "metadata.json"), "w"
+            ) as metadata_file:
+                json.dump(
+                    {"clusterName": self.cluster_name, "infraID": self.cluster_name},
+                    metadata_file,
+                )
+            # load API and Ingress IPs from config
+            self.api_vip = config.ENV_DATA["api_vip"]
+            self.ingress_vip = config.ENV_DATA["ingress_vip"]
+
+            # prepare required dnsmasq configuration (for PXE boot)
+            self.configure_dnsmasq_on_helper_vm()
+
+            # prepare ipxe directory in web document root
+            cmd = f"mkdir -m 755 -p {self.bm_config['bm_httpd_document_root']}/ipxe"
```

Comment:
The function in paramico library doesn't directly support the creation of parent directories, so it will make the function little bit more complex. I'll handle this prospectively in separated PR.